test/optimism/TokenRateNotifier.unit.test.ts

```diff
@@ -0,0 +1,126 @@
+import hre from "hardhat";
+import { assert } from "chai";
+import { unit } from "../../utils/testing";
+import { BigNumber, utils } from 'ethers'
+import { ethers } from 'hardhat'
+
+import {
+    TokenRateNotifier__factory,
+    ObserversArray__factory,
+    OpStackTokenRateObserver__factory,
+    ITokenRateObserver__factory,
+} from "../../typechain";
+
+unit("TokenRateNotifier", ctxFactory)
+
+  .test("init with wrong interface", async (ctx) => {
+    const { deployer } = ctx.accounts;
+    await assert.revertsWith(new ObserversArray__factory(deployer).deploy(BigNumber.from("0xffffffff")._hex), "ErrorInvalidInterface()");
+  })
+
+  .test("initial state", async (ctx) => {
+    const { tokenRateNotifier } = ctx.contracts;
+
+    assert.equalBN(await tokenRateNotifier.MAX_OBSERVERS_COUNT(), 16);
+    assert.equal(await tokenRateNotifier.INVALID_INTERFACE_ID(), "0xffffffff");
+    const iTokenRateObserver = getInterfaceID(ITokenRateObserver__factory.createInterface());
+    assert.equal(await tokenRateNotifier.REQUIRED_INTERFACE(), iTokenRateObserver._hex);
+    assert.equalBN(await tokenRateNotifier.observersLength(), 0);
+  })
+
+  .test("add zero address observer", async (ctx) => {
```

Comment:
(low priority) Consider improving tests names. E.g. include both condition and expected outcome like
- `add zero address observer` --> `revert on adding zero address observer`
- `happy path of adding observer`